app/src/main/java/io/intrepid/contest/screens/entrysubmission/entryimage/EntryImagePresenter.java

```diff
@@ -39,7 +43,28 @@ public void onBitmapReceived(Bitmap bitmap) {
 
     @Override
     public void onEntrySubmitted() {
-        Timber.d("Submit entry");
+        Timber.d("Entry creation API call.");
+
+        String contestId = persistentSettings.getCurrentContestId().toString();
+        EntryRequest entryRequest = new EntryRequest(entryName, bitmap);
+
+        Disposable disposable = restApi.createEntry(contestId, entryRequest)
+                .compose(subscribeOnIoObserveOnUi())
+                .subscribe(response -> showResult(response), throwable -> {
+                    Timber.d("API error: " + throwable.getMessage());
+                    view.showApiErrorMessage();
```

Comment:
If you'd like you can pass in a string resource into a single function you can too. I know it seems like you're putting an "Android" thing into a presenter but its just an integer so its allowed.